### DISCUSSION TITLE
How to check/add logs to the UI.

### DISCUSSION THREAD

**Discussion by PriyaranjanMaratheDish** [open-webui/open-webui#352] [Q&A]
I have deployed the UI 

Ollama is running on the EC2 - http://127.0.0.1:11434/

When I curl - I am able to see the following - 

*   Trying 127.0.0.1:11434...
* Connected to 127.0.0.1 (127.0.0.1) port 11434 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:11434
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Type: text/plain; charset=utf-8
< Date: Tue, 02 Jan 2024 21:01:19 GMT
< Content-Length: 17
<
* Connection #0 to host 127.0.0.1 left intact

I then built and hosted on Docker using the following 

docker run -d -p 8000:8080 --add-host=host.docker.internal:host-gateway -v ollama-webui:/app/backend/data --name ollama-webui --restart always ollama-webui

When I curl curl -v http://127.0.0.1:8000/
*   Trying 127.0.0.1:8000...
* Connected to 127.0.0.1 (127.0.0.1) port 8000 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8000
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< date: Tue, 02 Jan 2024 21:02:20 GMT
< server: uvicorn
< content-length: 22
< content-type: application/json
< x-process-time: 0
<
* Connection #0 to host 127.0.0.1 left intact

However how do I see docker logs for this app?

**Comment by PriyaranjanMaratheDish** [comment]
This is the error I get 

>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< date: Tue, 02 Jan 2024 22:06:47 GMT
< server: uvicorn
< content-length: 22
< content-type: application/json
< x-process-time: 0

**Comment by tjbck** [comment]
Hi! Have you tried our alternative method from our https://github.com/ollama-webui/ollama-webui/blob/main/TROUBLESHOOTING.md?

**Reply by PriyaranjanMaratheDish** [reply]
Same response - see below - 

root@ip-10-211-58-17:/var/snap/amazon-ssm-agent/7736# curl -v http://127.0.0.1:8080
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< date: Wed, 03 Jan 2024 03:29:05 GMT
< server: uvicorn
< content-length: 22
< content-type: application/json
< x-process-time: 0
<
* Connection #0 to host 127.0.0.1 left intact
{"detail":"Not Found"}root@ip-10-211-58-17:/var/snap/amazon-ssm-agent/7736#

**Reply by tjbck** [reply]
@PriyaranjanMaratheDish, Not entirely sure what I'm looking at 😅, Are you able to access Ollama WebUI? If you just want to check the docker container logs, use the following command: `docker logs ollama-webui`, Thanks!

**Reply by PriyaranjanMaratheDish** [reply]
Thanks. Logs are as follows - 

docker logs ollama-webui
INFO:     Started server process [7]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:     127.0.0.1:53518 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:50582 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:37594 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:41944 - "GET / HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:52848 - "GET / HTTP/1.1" 404 Not Found

Confirmed that the logs are updated on each curl - curl -v http://0.0.0.0:8080

**Reply by PriyaranjanMaratheDish** [reply]
Additional logs - 

ollama serve
Error: listen tcp 127.0.0.1:11434: bind: address already in use

I see one problem though - 

Not able to curl to the api correctly. Please see below - 

curl -v http://127.0.0.1:11434/api

*   Trying 127.0.0.1:11434...
* Connected to 127.0.0.1 (127.0.0.1) port 11434 (#0)
> GET /api HTTP/1.1
> Host: 127.0.0.1:11434
> User-Agent: curl/7.81.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Content-Type: text/plain
< Date: Wed, 03 Jan 2024 03:44:10 GMT
< Content-Length: 18
<
* Connection #0 to host 127.0.0.1 left intact

**Reply by tjbck** [reply]
Are you unable to access webui at all? Could you elaborate more on the issue you're facing? Thanks!

**Reply by PriyaranjanMaratheDish** [reply]
I have deployed the model and UI on a AWS EC2 Amazon Ubuntu Machine which doesn't have a browser for me to check the UI. I have setup a website to point to the 8080 port with Route 53, ALB etc. That is the reason I don't know if the UI is up till I can curl correctly on 8080.

**Reply by tjbck** [reply]
Not sure If I'm understanding everything but I'm guessing the core issue is you not being able to access the webui with the external EC2 instance URL?

**Reply by PriyaranjanMaratheDish** [reply]
Yes. I see two issues here. 

1)The API - http://127.0.0.1:11434/api doesn't work. Are there any additional steps for http://127.0.0.1:11434/api to work correcty?
2)WebUP is not able to connect to http://127.0.0.1:11434/api. I think #2 is because of #1. 

I will post http://127.0.0.1:11434/api issue in the main repo as well. It doesn't work on my mac and EC2 as well.

Thanks @tjbck

**Reply by PriyaranjanMaratheDish** [reply]
https://github.com/jmorganca/ollama/issues/1768 is the issue I opened. Hope to get it solved today.

**Comment by PriyaranjanMaratheDish** [comment]
> Yes. I see two issues here.
> 
> 1)The API - http://127.0.0.1:11434/api doesn't work. Are there any additional steps for http://127.0.0.1:11434/api to work correcty? 2)WebUP is not able to connect to http://127.0.0.1:11434/api. I think #2 is because of #1.
> 
> I will post http://127.0.0.1:11434/api issue in the main repo as well. It doesn't work on my mac and EC2 as well.
> 
> Thanks @tjbck
